### DISCUSSION TITLE
TFT in demo build

### DISCUSSION THREAD

**Discussion by ArchangelDesign** [PlummersSoftwareLLC/NightDriverStrip#23] [Q&A] [answered]
I'm trying to compile `demo` with USE_TFT enabled and either get missing `U8G2_SSD1306_128X64_NONAME_F_HW_I2C` in `main.cpp` or (when I add U8G2 library)
```Compiling .pio\build\demo\src\drawing.cpp.o
Compiling .pio\build\demo\src\effects.cpp.o

cc1plus.exe: out of memory allocating 65536 bytes

cc1plus.exe: out of memory allocating 65536 bytes
*** [.pio\build\demo\src\drawing.cpp.o] Error 1
*** [.pio\build\demo\src\audio.cpp.o] Error 1
```

so I disabled webserver and now stuck with `src/screen.cpp:38:5: error: 'U8G2_SSD1306_128X64_NONAME_F_HW_I2C' does not name a type`

How can I get TFT to run in demo mode?

**Comment by davepl** [comment, marked as answer]
I guess I've never made demo compile with the TFT yet!  You can open an issue and assign to me, or fix directly!

**Reply by ArchangelDesign** [reply]
thanks, Dave, awesome job on this BTW, everything compiles and works on the first try. The code has a very nice smell of the '90s which I love. I'll submit a PR with TFT support, I have the strip behind my monitors and want to mount esp32dev at the bottom of my primary monitor. For that reason, I'd like to see something on it, plus have a physical button to interact (although I love the wifi approach)

**Comment by davepl** [comment]
If you have a HELTEC module, just set USE_TFT to 1.  If you have an M5, set USE_OLED to 1.  That’ll five you a display.  There’s a Button object in there now you could change to step through effects pretty easy!

**Comment by davepl** [comment]
Fixed tonight - you can now just set USE_TFT to 1 and if you have the TFT, it'll work!  The code was already there but the headers weren't in the right order to compile for that scenario, so I just had to move them around a bit.

**Reply by ArchangelDesign** [reply]
yup, now it works, pulling 120fps on heltec :)
